programs/drift/src/math/margin.rs

```diff
@@ -490,6 +574,28 @@ pub fn calculate_margin_requirement_and_total_collateral_and_liability_info(
             calculation.track_open_orders_fraction(),
         )?;
 
+        let fuel_base_asset_value = if context.fuel_perp_delta.is_some()
```

Comment:
nit: might be slightly easier t add:

```
        if let Some((market_index, perp_delta)) = context.fuel_perp_delta {
            if market_index == market.market_index {
                market_position
                    .base_asset_amount
                    .safe_add(perp_delta)?
                    .cast::<i128>()?
                    .safe_mul(oracle_price_data.price.cast()?)?
                    .safe_div(AMM_RESERVE_PRECISION_I128)?
                    .unsigned_abs()
            } else {
                base_asset_value
            }
        } else {
            base_asset_value
        }
```

Reply:
nit: we probably wanna skip this calculation if fuel_bonus is 0 on market?,

Reply:
maybe we should just encapsulate all this logic in the margin calculation and skip if the market fuel bonus is 0

Reply:
its already in `calculate_margin_requirement_and_total_collateral_and_liability_info` where else should it be consolidated re: "in the margin calculation".